### PR TITLE
simulate policy extended

### PR DESCRIPTION
**Pull request by andrey-smirnov**
SimulatePrincipalPolicy and SimulateCustomPolicy API calls extended to accept ResourceArns and ContextEntries arguments.

### CODE REVIEW

File: src/erlcloud_iam.erl [hunk 1]
```diff
@@ -707,26 +709,54 @@ get_credential_report(Config) ->
     iam_query(Config, "GetCredentialReport", [], ItemPath, DataTypeDef).
 
 simulate_principal_policy(PolicySourceArn, ActionNames) ->
-    simulate_principal_policy(PolicySourceArn, ActionNames, default_config()).
+    simulate_principal_policy(PolicySourceArn, ActionNames, [], [], default_config()).
 
-simulate_principal_policy(PolicySourceArn, ActionNames, #aws_config{} = Config)
-  when is_list(PolicySourceArn) andalso is_list(ActionNames) ->
+simulate_principal_policy(PolicySourceArn, ActionNames, #aws_config{} = Config) ->
```
Comment (motobob) [inline src/erlcloud_iam.erl:714]: please add specs for all these funs

File: src/erlcloud_iam.erl [hunk 2]
```diff
@@ -707,26 +709,54 @@ get_credential_report(Config) ->
...
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, [], default_config()).
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, #aws_config{} = Config) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, [], Config);
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries, default_config()).
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries, #aws_config{} = Config)
```
Comment (motobob) [inline src/erlcloud_iam.erl:726]: should we check that list of action is not empty? I think it will be valuable...

**Comment by motobob** [comment]
@andrey-smirnov please add at least one test which covers all new params (actions/resources/context_entries)

**Comment by kkuzmin** [comment]
+1 after adding specs, tests, guards.

**Comment by motobob** [comment]
please re-open when completed

**Pull request closed without merging**